Makefile

```diff
@@ -0,0 +1,131 @@
+install/gh-release/%:
+	@ echo "[$@]: Installing $*..."
+	$(call download_github_release,$(FILENAME),$(OWNER),$(REPO),$(VERSION),$(QUERY))
+	chmod +x $(FILENAME)
+	$* --version
+	@ echo "[$@]: Completed successfully!"
+
+zip/install:
+	@ echo "[$@]: Installing $(@D)..."
+	apt-get install zip -y
+	@ echo "[$@]: Completed successfully!"
+
+terraform/install: TERRAFORM_VERSION_LATEST := $(CURL) https://checkpoint-api.hashicorp.com/v1/check/terraform | jq -r -M '.current_version' | sed 's/^v//'
+terraform/install: TERRAFORM_VERSION ?= $(shell $(TERRAFORM_VERSION_LATEST))
+terraform/install: | $(BIN_DIR) guard/program/jq
+	@ echo "[$@]: Installing $(@D)..."
+	$(call download_hashicorp_release,$(@D).zip,$(@D),$(TERRAFORM_VERSION))
+	unzip $(@D).zip && rm -f $(@D).zip && chmod +x $(@D)
+	mv $(@D) "$(BIN_DIR)"
+	$(@D) --version
+	@ echo "[$@]: Completed successfully!"
+
+terraform-docs/install: TFDOCS_VERSION ?= latest
+terraform-docs/install: | $(BIN_DIR) guard/program/jq
+	@ $(MAKE) install/gh-release/$(@D) FILENAME="$(BIN_DIR)/$(@D)" OWNER=segmentio REPO=$(@D) VERSION=$(TFDOCS_VERSION) QUERY='.name | endswith("$(OS)-$(ARCH)")'
+
+jq/install: JQ_VERSION ?= latest
+jq/install: | $(BIN_DIR)
+	@ $(MAKE) install/gh-release/$(@D) FILENAME="$(BIN_DIR)/$(@D)" OWNER=stedolan REPO=$(@D) VERSION=$(JQ_VERSION) QUERY='.name | endswith("$(OS)64")'
+
+shellcheck/install: SHELLCHECK_VERSION ?= latest
+shellcheck/install: SHELLCHECK_URL ?= https://storage.googleapis.com/shellcheck/shellcheck-${SHELLCHECK_VERSION}.linux.x86_64.tar.xz
+shellcheck/install: $(BIN_DIR) guard/program/xz
+	$(CURL) $(SHELLCHECK_URL) | tar -xJv
+	mv $(@D)-*/$(@D) $(BIN_DIR)
+	rm -rf $(@D)-*
+	$(@D) --version
+
+terraform/lint: | guard/program/terraform
+	@ echo "[$@]: Linting Terraform files..."
+	terraform fmt -check=true -diff=true
+	@ echo "[$@]: Terraform files PASSED lint test!"
+
+terraform/format: | guard/program/terraform
+	@ echo "[$@]: Formatting Terraform files..."
+	terraform fmt -recursive
+	@ echo "[$@]: Successfully formatted terraform files!"
+
+sh/%: FIND_SH := find . $(FIND_EXCLUDES) -name '*.sh' -type f -print0
+sh/lint: | guard/program/shellcheck
+	@ echo "[$@]: Linting shell scripts..."
+	$(FIND_SH) | $(XARGS) shellcheck {}
+	@ echo "[$@]: Shell scripts PASSED lint test!"
+
+json/%: FIND_JSON := find . $(FIND_EXCLUDES) -name '*.json' -type f
+json/lint: | guard/program/jq
+	@ echo "[$@]: Linting JSON files..."
+	$(FIND_JSON) | $(XARGS) bash -c 'cmp {} <(jq --indent 4 -S . {}) || (echo "[{}]: Failed JSON Lint Test"; exit 1)'
+	@ echo "[$@]: JSON files PASSED lint test!"
+
+json/format: | guard/program/jq
+	@ echo "[$@]: Formatting JSON files..."
+	$(FIND_JSON) | $(XARGS) bash -c 'echo "$$(jq --indent 4 -S . "{}")" > "{}"'
+	@ echo "[$@]: Successfully formatted JSON files!"
+
+tfdocs-awk/install: $(BIN_DIR)
+tfdocs-awk/install: ARCHIVE := https://github.com/plus3it/tfdocs-awk/archive/master.tar.gz
```

Comment:
We need to make an upstream change so that this is pinnned

Reply:
updated here: https://github.com/plus3it/cookiecutter-tfmodule/pull/29